### PR TITLE
DS-3102 Add checks to clients_first_seen_v2

### PR DESCRIPTION
**Pull request by ANich**
See https://mozilla-hub.atlassian.net/browse/DS-3102

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-2664)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/checks.sql [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    daily.client_id
+  FROM
+    telemetry_derived.clients_daily_v6 daily
+  LEFT JOIN
+    telemetry_derived.clients_first_seen_v2 first_seen
+    USING (client_id)
+  WHERE
+    submission_date = @submission_date
+    AND first_seen.client_id IS NULL
+)
```
Comment (scholtzan) [inline sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/checks.sql:16]:
```suggestion
);
```

Comment (scholtzan) [reply sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/checks.sql:16]: I thought something might be up with the new `generate-sql` task changes, but the semicolon should fix the task failure

File: sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/checks.sql [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#fail
+{{ is_unique(columns=["client_id"]) }}
+
+#fail
+ASSERT(
+  SELECT
+    COUNTIF(first_seen.client_id IS NULL)
+  FROM
+    `{{ project_id }}.{{ dataset_id }}.clients_daily_v6` daily
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/checks.sql:9]:
```suggestion
    `{{ project_id }}.{{ dataset_id }}.clients_daily_v6` AS daily
```

File: sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/checks.sql [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  LEFT JOIN
+    `{{ project_id }}.{{ dataset_id }}.{{ table_name }}` first_seen
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/checks.sql:11]:
```suggestion
    `{{ project_id }}.{{ dataset_id }}.{{ table_name }}` AS first_seen
```

File: sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/checks.sql [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+) = 0;
+
+#fail
+ASSERT(
+  SELECT
+    COUNTIF(first_seen.client_id IS NULL)
+  FROM
+    `{{ project_id }}.telemetry.new_profile`
+  LEFT JOIN
+    `{{ project_id }}.{{ dataset_id }}.{{ table_name }}` first_seen
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/checks.sql:24]:
```suggestion
    `{{ project_id }}.{{ dataset_id }}.{{ table_name }}` AS first_seen
```

File: sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/checks.sql (hunk 2 above)
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/checks.sql:9]: Is this check meant to ensure all clients in the daily table have an entry in our table?

File: sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/checks.sql [hunk 5]
```diff
@@ -0,0 +1,32 @@
+#fail
+{{ is_unique(columns=["client_id"]) }}
+
+#fail
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/checks.sql:4]: Should this be a fail and stop ETL execution?

**Review by kik-kik — APPROVED** [review]

File: sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/checks.sql (hunk 2 above)
Comment (ANich) [reply sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/checks.sql:9]: Yup, and similarly for the new profile ping table

File: sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/checks.sql (hunk 5 above)
Comment (ANich) [reply sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/checks.sql:4]: Yeah if there's a missing client something has gone wrong

**Pull request merged**